apis/nucleus/src/components/listbox/useSelectionsInteractions.js

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState, useCallback } from 'react';
+import { getUniques, selectValues, applySelectionsOnPages } from './listbox-selections';
+
+export default function useSelectionsInteractions({ layout, selections, pages = [], doc = document }) {
+  const [mouseDown, setMouseDown] = useState(false);
+  const [selectedElementNumbers, setSelectedElementNumbers] = useState([]);
+  const [selectingValues, setSelectingValues] = useState(false);
+  const [instantPages, setInstantPages] = useState(pages);
+
+  const select = ({ elemNumbers }) => {
+    setSelectingValues(true);
+    const isSingleSelect = layout.qListObject.qDimensionInfo.qIsOneAndOnlyOne;
+    return selectValues({ selections, elemNumbers, isSingleSelect }).then(() => {
+      setSelectingValues(false);
+    });
+  };
+
+  const onMouseDown = useCallback(
```

Comment:
Consider moving some logic inside onMouseDown. (direct or by function call)
instead of triggering adittional useEffects (I hope it would make the code easier to read)

Reply:
Yes, it looks a bit unmotivated for simple selections. I think this will make more sense when adding the code for range selections since it will then divide the handling of performing selections vs preparing for selections. But I will keep that in mind for the next PR.